conans/test/functional/generators/virtualenv_windows_bash_test.py

```diff
@@ -0,0 +1,128 @@
+                
+                # Add something to the path
+                self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
+                
+    """)
+
+    @classmethod
+    def setUpClass(cls):
+        cache_folder = os.path.join(temp_folder(path_with_spaces=False), ".conan")
+        t = TestClient(cache_folder=cache_folder)
+        t.save({"conanfile.py": cls.conanfile})
+        t.run("create . name/version@")
+
+        cls.cache_folder = os.path.dirname(t.cache_folder)
+
+    def _run_environment(self):
+        stdout, _ = subprocess.Popen(["where", "conan"], stdout=subprocess.PIPE).communicate()
+        conan_path = decode_text(stdout).splitlines()[0]  # Get the first one (Windows returns all found)
+
+        # All the commands are listed in a sh file:
+        commands_file = os.path.join(self.test_folder, 'commands.sh')
+        with open(commands_file, 'w+') as f:
+            # Dirty environment
+            f.write("export USER_VAR=existing_value\n")
+            f.write("export ANOTHER=existing_value\n")
+            f.write("export WHATEVER=existing_value\n")
+            f.write("export WHATEVER2=existing_value\n")
+            f.write("export CFLAGS=existing_value\n")
+
+            f.write("export PATH={}:$PATH\n".format(os.path.dirname(conan_path).replace('\\', '/').replace('C:', '/c')))
+            f.write("export CONAN_USER_HOME={}\n".format(self.cache_folder.replace('\\', '/').replace('C:', '/c')))
+            f.write("conan install name/version@ -g virtualenv -g virtualrunenv\n")
+
+            f.write("env > env_before.txt\n")
+            f.write("echo 'Start to find executable'\n")
+            f.write("echo __exec_pre_path__=$(which executable)\n")
+            f.write(". ./activate.sh\n")
+            f.write("env > env_activated.txt\n")
+            f.write("echo __exec_env_path__=$(which executable)\n")
+            f.write("executable\n")
+            f.write(". ./deactivate.sh\n")
+            f.write("echo __exec_post_path__=$(which executable)\n")
+            f.write("env > env_after.txt\n")
+
+        cmd = r'C:\Windows\System32\cmd.exe /c ""C:\Program Files\Git\bin\sh.exe" --login -i "{}""'.format(commands_file)
+        shell = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=self.test_folder)
+        (stdout, stderr) = shell.communicate()
+        stdout, stderr = decode_text(stdout), decode_text(stderr)
+
+        env_before = _load_env_file(os.path.join(self.test_folder, "env_before.txt"))
+        env_after = _load_env_file(os.path.join(self.test_folder, "env_after.txt"))
+        self.assertDictEqual(env_before, env_after)  # Environment restored correctly
+
+        return stdout, _load_env_file(os.path.join(self.test_folder, "env_activated.txt"))
+
+    def test_basic_variable(self):
+        self.test_folder = temp_folder(path_with_spaces=False)
+        _, environment = self._run_environment()
```

Comment:
I am a bit confused. These 3 tests looks that they are running exactly the same environment execution, and they just differ in the checks. In that case, I really think that it should be just 1 test, with all the necessary checks, and also maybe the setUpClass could be simplified.